corehq/apps/accounting/management/commands/upgrade_software_plan_version.py

```diff
@@ -0,0 +1,96 @@
+import logging
+
+from django.core.management import BaseCommand
+
+from django_prbac.models import Role
+
+from corehq.apps.accounting.models import SoftwarePlanVersion
+
+logger = logging.getLogger(__name__)
+
+
+class OldRoleDoesNotExist(Exception):
+    pass
+
+
+class NewRoleDoesNotExist(Exception):
+    pass
+
+
+def upgrade_software_plan_version(old_role, new_role, limit_to_plans=None, dry_run=False):
+    """
+    :param old_role: slug for role to search for
+    :param new_role: slug for role that new software plan version should reference
+    :param limit_to_plans: optionally limit to specific plan names
+    :param dry_run: if False, will make changes to the DB
+    :return: a list of plan names that were upgraded
+    """
+    dry_run_tag = '[DRY_RUN]' if dry_run else ''
+
+    try:
+        Role.objects.get(slug=old_role)
+    except Role.DoesNotExist:
+        raise OldRoleDoesNotExist
+
+    try:
+        new_role_obj = Role.objects.get(slug=new_role)
+    except Role.DoesNotExist:
+        raise NewRoleDoesNotExist
+
+    versions = SoftwarePlanVersion.objects.filter(role__slug=old_role, is_active=True)
+    if limit_to_plans:
+        versions = versions.filter(plan__name__in=limit_to_plans)
+
+    upgraded_plans = set()
+    for old_version in versions:
+        new_version = SoftwarePlanVersion(
+            plan=old_version.plan,
+            product_rate=old_version.product_rate,
```

Comment:
don't forget about `feature_rates` too, that's really important!

Reply:
Yeah it seemed like I had to save the SoftwarePlanVersion first before setting the feature_rates field, which [I do below](https://github.com/dimagi/commcare-hq/pull/30988/files/63c61660ae7c4eabd611709639ebc763dc14cbe9..e3f5458f4bc53abf2293e7ed58a5d38505de492f#diff-e30a5fe18fb78909de5bfade76650f62514e8e5c715efc79333827a4544f958bR54).